hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcTable.java

```diff
@@ -131,6 +127,10 @@ public SerializationService getSerializationService() {
         return serializationService;
     }
 
+    public DataConnectionService getDataConnectionService() {
```

Comment:
should be protected, getting DataConnectionService from here would be at best a workaround

Reply:
I see you use it. It is not a responsibility of `JdbcTable` to pass here and there `DataConnectionService`

Reply:
That's a good point. It seems that the serialization service is not used at all, I wonder where that came from. I will remove it.

However, I need the DataConnectionService in the methods returning processors so I can build a correct query according to the dialect. And the query can't be built in the PMS, that's too late - it wouldn't be cached. We don't want to rebuild the query each time it is issued.

I wonder if adding `NodeEngine` to DagBuildContext would be acceptable. It is something I use to build the DAG after all. 
We could easily pass it here

https://github.com/hazelcast/hazelcast/blob/8f03413cad9624995cfdba61ea937cd55f1ac531/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateTopLevelDagVisitor.java#L123

WDYT? cc @Fly-Style

Reply:
I think we can add NodeEngine to DagBuildContext, so objects that should be simple remain simple

Reply:
I forgot that one day `SqlConnector` and `DagBuildContext` will become public API. `NodeEngine` is not public but `DataConnectionService` is. It would be better to have the latter one in public API, but as for now we can leave it as is.